syntaxes/yml.tmLanguage.YAML-tmLanguage

```diff
@@ -200,21 +153,38 @@ repository:
                 match: (\+|-|/|mod|\*)
                 name: keyword.operator.mathematics.yseop
                 comment: 'Some of the operators that can be find in YML.'
-    functions:
-        begin: \b[\p{L}_\p{N}]+(::?[\p{L}_\p{N}]+)*\s*\(
+    method-calls:
+        begin: >
+            (?x)
+            (\.)
+            ([\p{L}_\p{N}]+(::?[\p{L}_\p{N}]+)*)
+            \s*
+            (\()
+        beginCaptures:
+            '1': { name: punctuation.separator.period.yseop }
+            '2': { name: entity.name.function.yseop }
+            '4': { name: punctuation.definition.parameters.begin.bracket.round.yseop }
         end: \)
+        endCaptures:
+            '0': { name: punctuation.definition.parameters.end.bracket.round.yseop }
+        name: meta.method-call.yseop
         patterns:
-            -
-                begin: \b
-                match: ([\p{L}_\p{N}]+|\d+)
-                end: \b
-                name: variable.parameter.yseop
-                comment: 'A parameter is just a string or a number'
-            -
-                include: '#strings'
-            -
-                include: '#constants'
-            -
-                include: '#comments'
-        name: entity.name.function.yseop
-        comment: 'A function call in YML.'
+            - {include: '$self'}
+    function-calls:
+        begin: >
+            (?x)
+            # Keywords that have arguments and may be wrongfully found as functions.
+            (?!\\?(case|for|foreach|if|ifExpr|switch|switchExpr|switchExprExclusive|while))
+            (\\?[\p{L}_\p{N}]+(::?[\p{L}_\p{N}]+)*)
+            \s*
+            (\()
+        beginCaptures:
+            '2': { name: entity.name.function.yseop }
+            # '3': { name: entity.name.function.yseop }
```

Comment:
This line is commented out, should it stay that way or be removed ?

Reply:
Oh. I should remove it. You're right.